src/ol_infrastructure/components/services/vault.py

```diff
@@ -199,24 +210,29 @@ def __init__(
             approle_config: OLVaultAppRoleAuthBackendConfig,
             opts: ResourceOptions = None
     ):
-        super().__init__('ol:services:VaultAppRoleAuthBackend', approle_config.name, None, opts)
+        super().__init__('ol:services:VaultAppRoleAuthBackend', approle_config.authbackend_name, None, opts)
```

Comment:
Just a nitpick, but since we have a number of components for Vault we should probably change the namespacing to e.g. `ol:services:Vault:AppRoleAuthBackend`